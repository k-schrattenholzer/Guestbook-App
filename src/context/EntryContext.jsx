import { useContext, createContext, useState } from 'react'

// Create the Context
const EntryContext = createContext();

// Create the EntryProvider component
const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}</EntryContext.Provider>
  )
}

// Create custom hook

 // `context` will be undefined when using this custom hook
  // outside of the provider, so we throw an error 

  // Export the provider component and the custom hook