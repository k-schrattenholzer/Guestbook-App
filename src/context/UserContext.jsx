// Create the Context
// Create the UserProvider component
// Create custom hook

 // `context` will be undefined when using this custom hook
  // outside of the provider, so we throw an error 

  // Export the provider component and the custom hook