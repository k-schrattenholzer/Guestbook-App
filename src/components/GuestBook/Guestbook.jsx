import { useState } from 'react';
import { useEntries } from '../../context/EntryContext';
import { useUser } from '../../context/UserContext';
import './GuestBook.css';

export default function Guestbook() {
  const [name, setName] = useState('');
  const [guestEntry, setGuestEntry] = useState('');
  const { entries, setEntries } = useEntries();
  const { user, setUser } = useUser();

  const guestNameInput = (
    <div>
      <label htmlFor="guestName">Visitor Name</label>
      <input
        className="guestNameInput"
        id="guestName"
        placeholder="what do they call ye"
        value={name}
        onChange={(e) => setName(e.target.value)}
        type="text"
      />
    </div>
  );

  const guestBookMessage = user
    ? `So interesting, ${name}.`
    : `What's your deal?`;

  return (
    <div>
      <h1>{guestBookMessage}</h1>
      <form onSubmit={handleSubmit}>
        {user ? null : guestNameInput}
        <div>
          <div>
            <label htmlFor="guestEntry">Guest Entry</label>
          </div>
          <div>
            <textarea
              id="guestEntry"
              value={guestEntry}
              placeholder="whaddya got noodlin around up there?"
              onChange={(e) => setGuestEntry(e.target.value)}
            />
          </div>
        </div>
        <div>
          <button type="submit">
            stick it on the wall
          </button>
          {user && (
            <button
              type='button'
              onClick={() => {
                setUser('');
                setName('');
              }}>
              if you're not {user}, you gotta go
            </button>
          )}
        </div>
      </form>
    </div>
  );
}
// TO DOS
// import useEntries and useUser
// destructure state and setState before render
// the name and the setName, and guestEntry and setGuestEntry will be kept locally
//will need an update guest name function
// and a handle submit that updates the guests name in state
// declare the HTML for the guest name input as a variable, and conditionally show
