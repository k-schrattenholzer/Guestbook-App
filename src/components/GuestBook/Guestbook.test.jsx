import { render, screen, waitFor } from "@testing-library/react";
// import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { EntryProvider } from "../../context/EntryContext.jsx";
import { UserProvider } from "../../context/UserContext.jsx";
import Guestbook from "./Guestbook.jsx";

// const mockLogout = jest.fn();

// jest.mock('../../hooks/useAuth', () => {
//   return {
//     useAuth: function () {
//       return {
//         user: { username: 'k@k.com' },
//         logout: mockLogout,
//       }
//     }
//   }
// });

it('should render the guestbook, when a user is logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <UserProvider>
        <EntryProvider>
          <Guestbook />
        </EntryProvider>
      </UserProvider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
})
