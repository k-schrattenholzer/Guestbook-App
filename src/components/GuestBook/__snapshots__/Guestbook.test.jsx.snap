// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the guestbook, when a user is logged in 1`] = `
<div>
  <div>
    <h1>
      What's your deal?
    </h1>
    <form>
      <div>
        <div>
          <label
            for="userEntry"
          >
            whaddya got noodlin around up there?
          </label>
        </div>
        <div>
          <textarea
            id="userEntry"
            placeholder="tell us everything..."
          />
        </div>
      </div>
      <div>
        <button
          type="submit"
        >
          stick it on the wall
        </button>
      </div>
    </form>
    
  </div>
</div>
`;
